ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx

```diff
@@ -130,122 +102,121 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
-    const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
-      { actions }
-    );
-    const addItemSpy = sinon.stub(actions, "addItem");
+    selector?.click();
 
-    const addRuleButton = getByTestId("rule-group-add-rule");
-    addRuleButton.click();
+    expect(
+      await findByText(
+        TestUtils.i18n.getLocalizedString(
+          "Components:filterBuilder.operators.and"
+        )
+      )
+    ).to.not.be.null;
 
-    expect(addItemSpy).to.be.calledOnceWith(defaultProps.path, "RULE");
+    expect(setRuleGroupOperatorSpy).to.be.calledWith(
+      defaultProps.path,
+      PropertyFilterRuleGroupOperator.Or
+    );
   });
 
-  it("dispatches add rule group event when button is clicked", () => {
+  it("Toggles operator 'Or' to 'And'", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroup,
+      path: [],
+      isGroupOperatorDisabled: false,
+    };
+    const { container, findByText } = renderWithContext(
+      <PropertyFilterBuilderRuleGroupRenderer {...props} />,
       { actions }
     );
-    const addItemSpy = sinon.stub(actions, "addItem");
+    const setRuleGroupOperatorSpy = sinon.stub(actions, "setRuleGroupOperator");
 
-    const addRuleGroupButton = getByTestId("rule-group-add-rule-group");
-    addRuleGroupButton.click();
+    const selector = container.querySelector<HTMLAnchorElement>(
+      ".fb-group-operator .iui-anchor"
+    );
+    expect(selector).to.not.be.null;
 
-    expect(addItemSpy).to.be.calledOnceWith(defaultProps.path, "RULE_GROUP");
+    selector?.click();
+
+    expect(
+      await findByText(
+        TestUtils.i18n.getLocalizedString(
+          "Components:filterBuilder.operators.or"
+        )
+      )
+    ).to.not.be.null;
+
+    expect(setRuleGroupOperatorSpy).to.be.calledWith(
+      defaultProps.path,
+      PropertyFilterRuleGroupOperator.And
+    );
   });
 
-  it("dispatches remove item event when button is clicked", () => {
+  it("'Or' Operator should not be clickable if toggled disabled", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer
-        {...defaultProps}
-        group={{
-          id: "id",
-          groupId: "parentGroupId",
-          operator: PropertyFilterRuleGroupOperator.And,
-          items: [],
-        }}
-      />,
+    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroup,
```

Comment:
Do not mutate `rootGroup` in test as it is shared between tests.
```suggestion
    const props: PropertyFilterBuilderRuleGroupRendererProps = {
      group: { ...rootGroup, operator: PropertyFilterRuleGroupOperator.Or },
```